src/Microsoft.OData.Core/Uri/ODataUriUtils.cs

```diff
@@ -95,15 +95,29 @@ public static object ConvertFromUriLiteral(string value, ODataVersion version, I
         /// </summary>
         /// <param name="value">Value to be converted.</param>
         /// <param name="version">Version to be compliant with.</param>
+        /// <param name="isIeee754Compatible">true if value should be IEEE 754 compatible.</param>
+        /// <returns>A string representation of <paramref name="value"/> for use in a Url.</returns>
+        [SuppressMessage("Microsoft.Design", "CA1055:UriReturnValuesShouldNotBeStrings", Justification = "designed to aid the creation on a URI, not create a full one")]
+        public static string ConvertToUriLiteral(object value, ODataVersion version, bool isIeee754Compatible)

```

Comment:
>bool isIeee754Compatible [](start = 85, length = 24)

Can we consider to create a "ConvertSetting" class and add "IsIeee754Compatible" as one of this setting.

The advantage is that, in the future if we have more configuration, we don't need to change the syntax of the method, we can just modify the ConvertSetting ?

Reply:
@odero an enum would restrict the type of settings. Remember a good design requires an enum to hold related values like days in a week. In this case we are bound to have a number of varied types/values.

Reply:
@xuzhg Do we see an increase in these configurations in future? At the moment we have very few configs for the function.

Reply:
So far no, :)


---
In reply to: [383678255](https://github.com/OData/odata.net/pull/1659#discussion_r383678255) [](ancestors = 383678255)

Reply:
@paulodero  I think a settings object would also be a better idea even if we only have one option. The reason I think so is because `IsIeee754Compatible` accounts for special case, applies only for numbers. In that sense, it doesn't seem like a "main" option and seem more suited to be part of a larger settings object than standalone parameter.

Reply:
Will be good to create an object for it if we have more settings.